### PR TITLE
delete unused PFBlockElementTrack

### PR DESCRIPTION
**Pull request by kpedro88**
Some 930 relval workflows crashed with bus errors, indicating a memory leak. See e.g. http://cms-unified.web.cern.ch/cms-unified/joblogs/sandhya_RVCMSSW_9_3_0TTbar_14TeV_Timing_PU25ns__2023D17PU140_170918_163355_2603/135/RecoFullGlobalPU_Timing_2023D17PU/5dc3710e-9d83-11e7-9242-02163e00d7b3-0-2-logArchive/cmsRun1/.

Running valgrind (on 0PU equivalent workflow), I found a memory leak in PF. This seems to be the culprit, comparing MemoryCheck output for the failing workflow above for 10 events before:
```
Begin processing the 10th record. Run 1, Event 2510, LumiSection 26 at 26-Sep-2017 18:35:31.780 CDT
MemoryCheck: module MixingModule:mix VSIZE 10413.9 0.390625 RSS 7344.68 -528.73
```
and after:
```
Begin processing the 10th record. Run 1, Event 2510, LumiSection 26 at 27-Sep-2017 10:28:38.319 CDT
MemoryCheck: module MixingModule:mix VSIZE 8084.34 0.390625 RSS 5014.22 -532.316
```

attn: @jnsandhya @slava77

### CODE REVIEW

**Comment by kpedro88** [comment]
please test

**Comment by kpedro88** [comment]
please test

**Comment by slava77** [comment]
Is the fix/leak phase-2 releated?
AFAIK, there are no significant leaks in phase-1/run2 reco at least since the last time this code was changed.

**Comment by kpedro88** [comment]
@slava77 `GeneralTracksImporterWithVeto` is only used in Phase2.

**Comment by slava77** [comment]
OK, thanks.
This explains.

File: RecoParticleFlow/PFProducer/plugins/importers/GeneralTracksImporterWithVeto.cc [hunk 1]
```diff
@@ -134,7 +134,7 @@ importToBlock( const edm::Event& e,
   }
   // now we actually insert tracks, again tagging muons along the way
   reco::PFRecTrackRef pftrackref;  
-  reco::PFBlockElementTrack* trkElem = NULL;
+  reco::PFBlockElementTrack* trkElem = nullptr;
```
Comment (slava77) [inline RecoParticleFlow/PFProducer/plugins/importers/GeneralTracksImporterWithVeto.cc]: can this be changed to a smart pointer instead?

Comment (kpedro88) [reply RecoParticleFlow/PFProducer/plugins/importers/GeneralTracksImporterWithVeto.cc]: @slava77 the value here is emplaced into a vector of smart pointers, so it is managed properly when it's used.

I'm trying to make minimal changes given the press of time.

**Comment by kpedro88** [comment]
@smuzaffar did a machine crash or something? The relval summary logs seem to be truncated.

**Comment by kpedro88** [comment]
Here are memory reports for 40 events in the failing workflow.

Before:
```
MemoryReport> Peak virtual size 13004.8 Mbytes
 Key events increasing vsize: 
[5] run: 1 lumi: 26 event: 2505  vsize = 9293.23 deltaVsize = 232 rss = 7553.57 delta = -58.6875
[2] run: 1 lumi: 26 event: 2502  vsize = 8660.53 deltaVsize = 298.277 rss = 7612.26 delta = 132.918
[8] run: 1 lumi: 26 event: 2509  vsize = 10173.5 deltaVsize = 416.25 rss = 7983.69 delta = 371.43
[17] run: 1 lumi: 26 event: 2516  vsize = 11549.9 deltaVsize = 408 rss = 8000.38 delta = 16.6875
[14] run: 1 lumi: 26 event: 2514  vsize = 11141.9 deltaVsize = 248 rss = 7796.24 delta = -187.453
[34] run: 1 lumi: 26 event: 2533  vsize = 12813.8 deltaVsize = 0 rss = 7734.79 delta = -107.496
[33] run: 1 lumi: 26 event: 2534  vsize = 12813.8 deltaVsize = 128 rss = 7842.28 delta = 82.8828
[39] run: 1 lumi: 26 event: 2540  vsize = 13004.8 deltaVsize = 191 rss = 7186.56 delta = -655.719
```

After:
```
MemoryReport> Peak virtual size 11142.5 Mbytes
 Key events increasing vsize: 
[3] run: 1 lumi: 26 event: 2504  vsize = 6627.88 deltaVsize = 244.699 rss = 5252.36 delta = -90.1719
[8] run: 1 lumi: 26 event: 2509  vsize = 7923.88 deltaVsize = 384 rss = 5684.53 delta = 432.164
[17] run: 1 lumi: 26 event: 2516  vsize = 9427.34 deltaVsize = 400 rss = 5647.23 delta = -37.3008
[35] run: 1 lumi: 26 event: 2535  vsize = 11139.5 deltaVsize = 191 rss = 5647.93 delta = 0.699219
[26] run: 1 lumi: 26 event: 2527  vsize = 10228.5 deltaVsize = 160.25 rss = 5421.34 delta = -225.891
[36] run: 1 lumi: 26 event: 2536  vsize = 11139.5 deltaVsize = 0 rss = 5397.52 delta = -250.402
[35] run: 1 lumi: 26 event: 2535  vsize = 11139.5 deltaVsize = 191 rss = 5647.93 delta = 328.539
[39] run: 1 lumi: 26 event: 2540  vsize = 11142.5 deltaVsize = 3 rss = 4849.57 delta = -798.359
```

**Comment by smuzaffar** [comment]
please test
@kpedro88 , looks like these tests ran on new cmsbuild machines and as there is no swap on these nodes to jobs were killed by system due to lack of resources. I have taken the news nodes offline for now. Tomorrow I will reconfigure them.

**Comment by slava77** [comment]
+1

for https://github.com/cms-sw/cmssw/pull/20669	f7052fd
- somewhat clear mem leak fix
- jenkins tests pass
- relying on observations by @kpedro88 in https://github.com/cms-sw/cmssw/pull/20669#issuecomment-332603468 [my valgrind run will take a little while to complete]

**Comment by davidlange6** [comment]
+1

**Pull request merged**